core/src/com/google/inject/internal/SingleMethodInjector.java

```diff
@@ -40,19 +41,21 @@ private MethodInvoker createMethodInvoker(final Method method) {
 
     /*if[AOP]*/
     try {
-      final net.sf.cglib.reflect.FastClass fastClass = BytecodeGen.newFastClassForMember(method);
-      if (fastClass != null) {
-        final int index = fastClass.getMethod(method).getIndex();
-
+      BiFunction<Object, Object[], Object> fastMethod = BytecodeGen.fastMethod(method);
+      if (fastMethod != null) {
         return new MethodInvoker() {
           @Override
           public Object invoke(Object target, Object... parameters)
               throws IllegalAccessException, InvocationTargetException {
-            return fastClass.invoke(index, target, parameters);
+            try {
+              return fastMethod.apply(target, parameters);
+            } catch (Throwable e) {
```

Comment:
(same q about wrapping w/ InvocationTargetException here)

Reply:
See above - we want to fall back to JDK reflection if any exception or class-loading linkage error occurs when creating the fast-method (note this doesn't cover the actual invocation.)

Reply:
See above